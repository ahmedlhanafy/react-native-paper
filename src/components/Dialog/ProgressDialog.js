/* @flow */
import * as React from 'react';
import { ActivityIndicator, View, StyleSheet } from 'react-native';
import Paragraph from '../Typography/Paragraph';
import Dialog from './Dialog';
import DialogTitle from './DialogTitle';
import DialogContent from './DialogContent';

type Props = {
  /**
   * Determines whether clicking outside the dialog dismiss it.
   */
  dismissable?: boolean,
  /**
   * Determines whether the dialog is visible.
   */
  visible: boolean,
  /**
   * Callback that is called when the user dismisses the dialog.
   */
  onDismiss: Function,
  /**
   * Size of the ActivityIndicator. It can take following values:
   * - for iOS: 'small' | 'large'
   * - for Android: number
   */
  size: 'small' | 'large' | number,
  /**
   * Color of the ActivityIndicator.
   */
  color?: string,
  /**
   * Title of the dialog.
   */
  title: string,
  /**
   * Color of the title.
   */
  titleColor?: string,
  /**
   * Text tha will be displayed next to ActivityIndicator.
   */
  text: string,
  /**
   * Color of the text.
   */
  textColor?: string,
  /**
   * Style that will be applied to whole Dialog e.g. backgroundColor
   */
  style?: any,
};

/**
 * Progress Dialog shows the user that application is processing something.
 *
 * ## Usage
 * ```js
 * import * as React from 'react';
 * import { Platform } from 'react-native';
 * import { Dialog } from 'react-native-paper';
 *
 * const isIOS = Platform.OS === 'ios';
 *
 * export default class MyComponent extends React.Component {
 *
 *   render() {
 *     return (
 *       <Dialog.ProgressDialog
 *         onDismiss={() => {}}
 *         size={isIOS ? 'large' : 48}
 *         text="Loading....."
 *         title="Progress Dialog"
 *       />
 *     );
 *   }
 * }
 * ```
 */
class ProgressDialog extends React.Component<Props> {
  static defaultProps = {
    dismissable: true,
    visible: false,
  };

  static displayName = 'Dialog.Progress';

  render() {
    const {
      dismissable,
      onDismiss,
      visible,
      style,
      title,
      titleColor,
      color,
      size,
      textColor,
      text,
    } = this.props;
    return (
      <Dialog
        dismissable={dismissable}
        onDismiss={onDismiss}
        visible={visible}
        style={style}
      >
        <DialogTitle style={{ color: titleColor }}>{title}</DialogTitle>
        <DialogContent>
          <View style={styles.content}>
            <ActivityIndicator
              color={color}
              size={size}
              style={{ marginRight: 16 }}
            />
            <Paragraph style={{ color: textColor }}>{text}</Paragraph>
          </View>
        </DialogContent>
      </Dialog>
    );
  }
}

const styles = StyleSheet.create({
  content: {
    flexDirection: 'row',
    alignItems: 'center',
  },
});

export default ProgressDialog;
