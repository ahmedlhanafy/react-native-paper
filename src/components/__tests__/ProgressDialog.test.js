/* @flow */

import * as React from 'react';
import renderer from 'react-test-renderer';
import ProgressDialog from '../Dialog/ProgressDialog';
import PortalHost from '../Portal/PortalHost';

const fn = () => {};

it('renders visible progress dialog', () => {
  const tree = renderer
    .create(
      <PortalHost>
        <ProgressDialog
          visible
          onDismiss={fn}
          size={48}
          text="Loading..."
          title="Progress Dialog"
        />
      </PortalHost>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders visible progress dialog with custom activity indicator color and style', () => {
  const tree = renderer
    .create(
      <PortalHost>
        <ProgressDialog
          visible
          onDismiss={fn}
          size="small"
          text="Loading..."
          title="Progress Dialog"
          color="#BADA55"
          style={{
            backgroundColor: '#999',
          }}
        />
      </PortalHost>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders not visible progress dialog', () => {
  const tree = renderer
    .create(
      <PortalHost>
        <ProgressDialog
          onDismiss={fn}
          size="small"
          text="Loading..."
          title="Progress Dialog"
        />
      </PortalHost>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
