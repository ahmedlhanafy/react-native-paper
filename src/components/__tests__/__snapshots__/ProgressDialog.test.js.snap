// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders not visible progress dialog 1`] = `
<View
  pointerEvents="box-none"
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;

exports[`renders visible progress dialog 1`] = `
<View
  pointerEvents="box-none"
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "backgroundColor": "rgba(0, 0, 0, .5)",
            },
          ]
        }
      />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        testID={undefined}
      />
      <View
        collapsable={undefined}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          collapsable={undefined}
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
              },
              Object {
                "backgroundColor": "#ffffff",
              },
              Object {
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 23,
                  "width": 0,
                },
                "shadowOpacity": 0.24,
                "shadowRadius": 24,
              },
              Object {
                "backgroundColor": "#ffffff",
                "borderRadius": 2,
                "elevation": 24,
                "marginHorizontal": 26,
                "marginVertical": 0,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "Helvetica Neue",
                },
                Array [
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "fontFamily": "HelveticaNeue-Medium",
                  },
                  Array [
                    Object {
                      "fontSize": 20,
                      "lineHeight": 30,
                      "marginVertical": 2,
                    },
                    Array [
                      Object {
                        "marginBottom": 18,
                        "marginHorizontal": 24,
                        "marginTop": 22,
                      },
                      Object {
                        "color": "#000000",
                      },
                      Object {
                        "color": undefined,
                      },
                    ],
                  ],
                  Array [
                    Object {
                      "fontSize": 20,
                      "lineHeight": 30,
                      "marginVertical": 2,
                    },
                    Array [
                      Object {
                        "marginBottom": 18,
                        "marginHorizontal": 24,
                        "marginTop": 22,
                      },
                      Object {
                        "color": "#000000",
                      },
                      Object {
                        "color": undefined,
                      },
                    ],
                  ],
                ],
              ]
            }
            theme={
              Object {
                "colors": Object {
                  "accent": "#ff4081",
                  "background": "#fafafa",
                  "disabled": "rgba(0, 0, 0, 0.26)",
                  "paper": "#ffffff",
                  "placeholder": "rgba(0, 0, 0, 0.38)",
                  "primary": "#3f51b5",
                  "text": "#000000",
                },
                "dark": false,
                "fonts": Object {
                  "light": "HelveticaNeue-Light",
                  "medium": "HelveticaNeue-Medium",
                  "regular": "Helvetica Neue",
                  "thin": "HelveticaNeue-Thin",
                },
                "roundness": 2,
              }
            }
          >
            Progress Dialog
          </Text>
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 24,
                  "paddingHorizontal": 24,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <ActivityIndicator
                animating={true}
                color="#999999"
                hidesWhenStopped={true}
                size={48}
                style={
                  Object {
                    "marginRight": 16,
                  }
                }
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "Helvetica Neue",
                    },
                    Array [
                      Object {
                        "color": "rgba(0, 0, 0, 0.87)",
                        "fontFamily": "Helvetica Neue",
                      },
                      Array [
                        Object {
                          "fontSize": 14,
                          "lineHeight": 20,
                          "marginVertical": 2,
                        },
                        Object {
                          "color": undefined,
                        },
                      ],
                      Array [
                        Object {
                          "fontSize": 14,
                          "lineHeight": 20,
                          "marginVertical": 2,
                        },
                        Object {
                          "color": undefined,
                        },
                      ],
                    ],
                  ]
                }
                theme={
                  Object {
                    "colors": Object {
                      "accent": "#ff4081",
                      "background": "#fafafa",
                      "disabled": "rgba(0, 0, 0, 0.26)",
                      "paper": "#ffffff",
                      "placeholder": "rgba(0, 0, 0, 0.38)",
                      "primary": "#3f51b5",
                      "text": "#000000",
                    },
                    "dark": false,
                    "fonts": Object {
                      "light": "HelveticaNeue-Light",
                      "medium": "HelveticaNeue-Medium",
                      "regular": "Helvetica Neue",
                      "thin": "HelveticaNeue-Thin",
                    },
                    "roundness": 2,
                  }
                }
              >
                Loading...
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders visible progress dialog with custom activity indicator color and style 1`] = `
<View
  pointerEvents="box-none"
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "backgroundColor": "rgba(0, 0, 0, .5)",
            },
          ]
        }
      />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        testID={undefined}
      />
      <View
        collapsable={undefined}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          collapsable={undefined}
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
              },
              Object {
                "backgroundColor": "#ffffff",
              },
              Object {
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 23,
                  "width": 0,
                },
                "shadowOpacity": 0.24,
                "shadowRadius": 24,
              },
              Object {
                "backgroundColor": "#999",
                "borderRadius": 2,
                "elevation": 24,
                "marginHorizontal": 26,
                "marginVertical": 0,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "Helvetica Neue",
                },
                Array [
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "fontFamily": "HelveticaNeue-Medium",
                  },
                  Array [
                    Object {
                      "fontSize": 20,
                      "lineHeight": 30,
                      "marginVertical": 2,
                    },
                    Array [
                      Object {
                        "marginBottom": 18,
                        "marginHorizontal": 24,
                        "marginTop": 22,
                      },
                      Object {
                        "color": "#000000",
                      },
                      Object {
                        "color": undefined,
                      },
                    ],
                  ],
                  Array [
                    Object {
                      "fontSize": 20,
                      "lineHeight": 30,
                      "marginVertical": 2,
                    },
                    Array [
                      Object {
                        "marginBottom": 18,
                        "marginHorizontal": 24,
                        "marginTop": 22,
                      },
                      Object {
                        "color": "#000000",
                      },
                      Object {
                        "color": undefined,
                      },
                    ],
                  ],
                ],
              ]
            }
            theme={
              Object {
                "colors": Object {
                  "accent": "#ff4081",
                  "background": "#fafafa",
                  "disabled": "rgba(0, 0, 0, 0.26)",
                  "paper": "#ffffff",
                  "placeholder": "rgba(0, 0, 0, 0.38)",
                  "primary": "#3f51b5",
                  "text": "#000000",
                },
                "dark": false,
                "fonts": Object {
                  "light": "HelveticaNeue-Light",
                  "medium": "HelveticaNeue-Medium",
                  "regular": "Helvetica Neue",
                  "thin": "HelveticaNeue-Thin",
                },
                "roundness": 2,
              }
            }
          >
            Progress Dialog
          </Text>
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 24,
                  "paddingHorizontal": 24,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <ActivityIndicator
                animating={true}
                color="#BADA55"
                hidesWhenStopped={true}
                size="small"
                style={
                  Object {
                    "marginRight": 16,
                  }
                }
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "Helvetica Neue",
                    },
                    Array [
                      Object {
                        "color": "rgba(0, 0, 0, 0.87)",
                        "fontFamily": "Helvetica Neue",
                      },
                      Array [
                        Object {
                          "fontSize": 14,
                          "lineHeight": 20,
                          "marginVertical": 2,
                        },
                        Object {
                          "color": undefined,
                        },
                      ],
                      Array [
                        Object {
                          "fontSize": 14,
                          "lineHeight": 20,
                          "marginVertical": 2,
                        },
                        Object {
                          "color": undefined,
                        },
                      ],
                    ],
                  ]
                }
                theme={
                  Object {
                    "colors": Object {
                      "accent": "#ff4081",
                      "background": "#fafafa",
                      "disabled": "rgba(0, 0, 0, 0.26)",
                      "paper": "#ffffff",
                      "placeholder": "rgba(0, 0, 0, 0.38)",
                      "primary": "#3f51b5",
                      "text": "#000000",
                    },
                    "dark": false,
                    "fonts": Object {
                      "light": "HelveticaNeue-Light",
                      "medium": "HelveticaNeue-Medium",
                      "regular": "Helvetica Neue",
                      "thin": "HelveticaNeue-Thin",
                    },
                    "roundness": 2,
                  }
                }
              >
                Loading...
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
